### DISCUSSION TITLE
Which queries are taking the most time

### DISCUSSION THREAD

**Discussion by vikaskedia** [CodeDredd/pinia-orm#720] [Q&A]
I am developing a large enterprise application using pinia-orm.

I am working on performance, I want to find out which queries are taking the most time inside the Pinia ORM engine.

Is there a way to ask Pinia-Orm to generate a table that will give me the total amount of time taken by the queries sorted by the time taken.

This way I can focus on the top most entries in this table and work on caching them

**Comment by CodeDredd** [comment]
Right now there is nothing for that. I need to think about it if such thing can be implemented. Do you know a js library which has such feature?

For now i would create a unit test which does that. If you want to write one you can look at mine for Inspiration 😉 
https://github.com/CodeDredd/pinia-orm/blob/master/packages/pinia-orm/tests/performance/save_has_many_relation.spec.ts
